trview.app/Geometry/IMesh.cpp

```diff
@@ -66,7 +66,8 @@ namespace trview
 
             if (offset_mode == SpriteOffsetMode::RoomSprite)
             {
-                offset = Vector3(0, (1 - object_height) * 0.5f, 0);
+                const float y_factor = (-0.5f + ((fabs(static_cast<float>(top)) / (bottom - top))));
```

Comment:
Worth a guard in case `bottom == top`? Just thinking of TRLE world.

Reply:
Good idea, not sure why they'd do that but good to avoid the div by 0